lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -480,6 +480,15 @@ void HostManager::updateExecutionStats(
                                 1000000 / duration);
 }
 
+void HostManager::ensureOutputsAvailable(ExecutionContext &context) {
```

Comment:
It's important to make sure we're calling this everywhere we use the HostManager until the user is able to deal with Tensors staying Device Resident. The big one I can think of is Onnxifi, but there may be others.